spec/features/user_show_spec.rb

```diff
@@ -0,0 +1,136 @@
+require 'rails_helper'
+
+RSpec.describe 'When I open user show page', type: :system do
+  before :all do
+    Like.delete_all
+    Comment.delete_all
+    Post.delete_all
+    User.delete_all
+    @first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                              bio: 'Teacher from Mexico.')
+    @second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                               bio: 'Teacher from Poland.')
+    @third_user = User.create(name: 'Alan', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                              bio: 'Singer from Mexico.')
+    Post.create(author: @first_user, title: 'Post1')
+    Post.create(author: @first_user, title: 'Post2')
+    Post.create(author: @first_user, title: 'Post3')
+    @latest_post = Post.create(author: @first_user, title: 'Post4')
+    Post.create(author: @second_user, title: 'Post1')
+    Post.create(author: @second_user, title: 'Post2')
+  end
+
+  it "shows the user's profile picture" do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_css("img[src='#{@first_user.photo}']")
+    visit users_path
+    sleep(1)
+    click_link(@second_user.name)
+    sleep(1)
+    expect(page).to have_css("img[src='#{@second_user.photo}']")
+    visit users_path
+    sleep(1)
+    click_link(@third_user.name)
+    sleep(1)
+    expect(page).to have_css("img[src='#{@third_user.photo}']")
+  end
+
+  it "shows the user's username" do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_content('Tom')
+    visit users_path
+    sleep(1)
+    click_link(@second_user.name)
+    sleep(1)
+    expect(page).to have_content('Lilly')
+    visit users_path
+    sleep(1)
+    click_link(@third_user.name)
+    sleep(1)
+    expect(page).to have_content('Alan')
+  end
+
+  it 'shows the number of posts the user has written' do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_content('Number of posts: 4')
+    visit users_path
+    sleep(1)
+    click_link(@second_user.name)
+    sleep(1)
+    expect(page).to have_content('Number of posts: 2')
+    visit users_path
+    sleep(1)
+    click_link(@third_user.name)
+    sleep(1)
+    expect(page).to have_content('Number of posts: 0')
+  end
```

Comment:
- [ ] [This](https://github.com/katarighe/blog-app/pull/9#discussion_r1384179722) requested change as well follows the same pattern as explained above.